src/utils/utils.ts

```diff
@@ -13,25 +13,27 @@ import input from "../input.json";
 import { PrimitiveType } from "../types/AttributeType";
 
 export const getOrCreateDomainModel = async (
-  model: IModel
+  model: IModel,
+  moduleName: string
 ): Promise<domainmodels.DomainModel> => {
-  return await getDomainModelInterface(model).load();
```

Comment:
In this function, you are only calling one other function. Can these be combined in one single function?